packages/devtools/lib/src/timeline/frame_flame_chart.dart

```diff
@@ -79,86 +83,91 @@ class FrameFlameChart extends CoreElement {
   }
 
   void _render(TimelineFrame frame) {
-    const int leftIndent = 80;
+    const int leftIndent = 70;
     const int rowHeight = 25;
 
-    // TODO(kenzie): re-write this scale logic.
-    const double microsPerFrame = 1000 * 1000 / 60.0;
-    const double pxPerMicro = microsPerFrame / 1000.0;
+    // 16,666 microseconds / frame will achieve a frame rate of 60 FPS.
+    const double targetMicrosPerFrame = 1000 * 1000 / 60.0;
 
-    int row = 0;
+    // Pixels per microsecond in order to fit a single frame in 1500px. In order
+    // for the whole frame to fit in 1500px at this drawing ratio, the frame
+    // duration must be [targetMicrosPerFrame] or less. 1500px is arbitrary.
+    const double pixelsPerMicro = 1500 / targetMicrosPerFrame;
 
-    final int microsAdjust = frame.startTime;
+    final int frameStartOffset = frame.startTime;
 
-    int maxRow = 0;
+    // Add 15 to account for vertical spacing between the CPU and GPU sections.
+    final cpuSectionHeight = frame.cpuEventFlow.getDepth() * rowHeight + 15;
+    final gpuSectionHeight = frame.gpuEventFlow.getDepth() * rowHeight;
+    final flameChartWidth = max(
+        element.clientWidth,
+        leftIndent +
+            (frame.gpuEventFlow.endTime - frame.cpuEventFlow.startTime) *
+                pixelsPerMicro);
 
-    void drawRecursively(TimelineEvent event, int row) {
-      final double start = (event.startTime - microsAdjust) / pxPerMicro;
-      final double end =
-          (event.startTime - microsAdjust + event.duration) / pxPerMicro;
+    void drawRecursively(TimelineEvent event, int row, CoreElement section) {
+      final double startPx =
+          (event.startTime - frameStartOffset) * pixelsPerMicro;
+      final double endPx = (event.endTime - frameStartOffset) * pixelsPerMicro;
 
       _drawFlameChartItem(
         event,
-        leftIndent + start.round(),
-        (end - start).round(),
+        leftIndent + startPx.round(),
```

Comment:
add a TODO that technically you  will want to round to fraction of a px
for high dpi devices where 1 logical pixel may equal  2 physical pixels, etc.

Reply:
done